### ISSUE TITLE
Resolve ambiguity in profile consumption best practices

### ISSUE THREAD

**Issue by Railboy** [microsoft/MixedRealityToolkit-Unity#5012]
## Describe the problem

There's no official best practice regarding profile data consumption or modification of profile data at runtime.

This has led to different services consuming profile data in different ways. Some variants:
- Copy all profile data locally and then discard the profile entirely.
- Copy mutable profile data locally, but continue to reference read-only profile data for lifetime of the service.
- Use profile data exclusively and make no local copies.

This has also led to profile inspectors behaving in different ways. Some allow you to modify their contents in play mode and observe the effects of those changes on the active service. Others are completely locked down.

Lots of open questions here. Should services copy and forget profile data? If not, should runtime profile usage be limited to read-only data? If not, is it ever appropriate to modify profile data at runtime to observe changes in service behavior? If so, _when_ is it appropriate (eg in editor vs in build)?

## Describe the solution you'd like

We need to define some profile data consumption best practices, then update our existing services to follow those best practices.

## Additional context

Discussed previously here: #4335

**Comment by davidkline-ms** [comment]
Agreed. Core systems and MRTK team provided extension services should behave the same with regards to data read from profiles.

I believe the most consistent behavior (shared by a majority of existing services) is the following:
- Read the profile into service properties
- Any profile settings that support runtime modification should have setters
- Read-only profile settings should have only getters
- All runtime changes should be made to the service properties, not the profile

**Comment by Zee2** [comment]
Closing as stale; I don't think we'll be making major changes to the profile systems in the future.
